sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -0,0 +1,426 @@
+   *
+   * The enclave then hands the byte buffer and the quote to its host. The host sends the quote and
+   * byte buffer as the "RunTime Data" to the via the {@link
+   * AttestationAsyncClient#attestSgxEnclave(BinaryData)}  or
+   * {@link AttestationAsyncClient#attestOpenEnclave} API. Assuming the byte buffer and quote are
+   * valid, and the quote contains the hash of the byte buffer, the attestation service responds
+   * with an {@link AttestationToken} signed by the attestation service, whose body is an {@link
+   * AttestationResult}.
+   *
+   * The token generated also includes the contents of the InitTimeData and/or RunTimeData if it was
+   * provided in the Attest API call.
+   *
+   * The host then sends the token to the relying party.  The relying party verifies the token
+   * and verifies the claims within the token indicate that the enclave is the correct enclave.
+   * It then takes the key from the token and uses it to encrypt the data to be sent to the
+   * enclave and sends that back to the host, which passes it into the enclave.
+   *
+   *
+   * That completes the secure key release protocol.
+   *
+   * When the Attestation Token is generated by the attestation service, as mentioned, it contains
+   * the InitTime and RunTime data.
+   *
+   * There are two possible representations for RunTime Data in the attestation token, depending on
+   * the requirements of the relying party:<br> The first is as JSON formatted data. That can be
+   * convenient if the relying party expects to receive its public key as a JSON Web Key The second
+   * is as a binary blob of data. That is needed if either the data sent by the enclave isn't a JSON
+   * object - for instance, if the RunTime data contained an asymmetric key which is formatted as a
+   * PEM encoded key, it should be interpreted as a binary blob
+   *
+   * If you ask for the RunTime data to be included in the token as binary, then it will be
+   * base64url encoded in the "x-ms-maa-enclavehelddata" claim in the output token (the
+   * {@link AttestationResult#getEnclaveHeldData()} property).
+   *
+   * If you ask for the RunTime data to be included in the token as JSON, then it will be included
+   * in the "x-ms-maa-runtimeClaims" claim in the output token (the {@link
+   * AttestationResult#getRuntimeClaims()} property).
+   *
+   * In addition to the Attest APIs, the {@link AttestationClient} object also contains helper APIs
+   * which can be used to retrieve the OpenId Metadata document and signing keys from the service.
+   *
+   *
+   * The OpenId Metadata document contains properties which describe the attestation service.
+   *
+   * The Attestation Signing Keys describe the keys which will be used to sign tokens generated by
+   * the attestation service. All tokens emitted by the attestation service will be signed by one
+   * of the certificates listed in the attestation signing keys.
+   *
+   */
```

Comment:
This will appear in the documentation for the AttestationClient object, and this documents the behavior of the AttestationClient object.

From discussions with the language architects, It's too heavyweight to be in the readme. I've not yet ported the readme over, but you can look at the [Attestation Client readme for Java](https://docs.microsoft.com/en-us/java/api/overview/azure/security-attestation-readme?view=azure-java-stable) to see where the readme will be coming from.